In [2]:
import numpy as np
import pandas as pd
from feature_creation import data_copy

# Create baseline performance using Naive model
## model = (always predict low demand)

In [3]:
data_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1600 entries, 0 to 1599
Data columns (total 19 columns):
 #   Column                 Non-Null Count  Dtype   
---  ------                 --------------  -----   
 0   hour_of_day            1600 non-null   category
 1   day_of_week            1600 non-null   category
 2   month                  1600 non-null   category
 3   holiday                1600 non-null   int64   
 4   weekday                1600 non-null   int64   
 5   summertime             1600 non-null   int64   
 6   temp                   1600 non-null   float64 
 7   dew                    1600 non-null   float64 
 8   humidity               1600 non-null   float64 
 9   precip                 1600 non-null   float64 
 10  snowdepth              1600 non-null   float64 
 11  windspeed              1600 non-null   float64 
 12  cloudcover             1600 non-null   float64 
 13  visibility             1600 non-null   float64 
 14  increase_stock         1600 non-null   c

In [4]:
data_copy.head()

hour_of_day day_of_week month  holiday  weekday  summertime  temp   dew  \
0           5           5     1        0        0           0  -7.2 -15.0   
1          21           4     1        0        1           0  -1.3 -12.8   
2          21           3     8        0        1           1  26.9  21.8   
3           1           6     1        0        0           0   3.1  -4.0   
4          17           0     3        0        1           0  11.7 -11.4   

   humidity  precip  snowdepth  windspeed  cloudcover  visibility  \
0     53.68     0.0        0.0       16.3        31.6        16.0   
1     40.97     0.0        0.0       23.9        85.7        16.0   
2     73.39     0.0        0.0        0.0        81.1        16.0   
3     59.74     0.0        0.0       19.2         0.0        16.0   
4     18.71     0.0        0.0       10.5        44.6        16.0   

    increase_stock  daytime  rushhour  weather_score  weather_score_daytime  
0  low_bike_demand        0         0      -0.141957              -0.000000  
1  low_bike_demand        0         0       0.075260               0.000000  
2  low_bike_demand        0         0       0.261261               0.000000  
3  low_bike_demand        0         0      -0.050008              -0.000000  
4  low_bike_demand        1         1       0.647846               0.647846

In [11]:
data_copy['y_hat'] = 'low_bike_demand'

In [12]:
data_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1600 entries, 0 to 1599
Data columns (total 20 columns):
 #   Column                 Non-Null Count  Dtype   
---  ------                 --------------  -----   
 0   hour_of_day            1600 non-null   category
 1   day_of_week            1600 non-null   category
 2   month                  1600 non-null   category
 3   holiday                1600 non-null   int64   
 4   weekday                1600 non-null   int64   
 5   summertime             1600 non-null   int64   
 6   temp                   1600 non-null   float64 
 7   dew                    1600 non-null   float64 
 8   humidity               1600 non-null   float64 
 9   precip                 1600 non-null   float64 
 10  snowdepth              1600 non-null   float64 
 11  windspeed              1600 non-null   float64 
 12  cloudcover             1600 non-null   float64 
 13  visibility             1600 non-null   float64 
 14  increase_stock         1600 non-null   c

In [16]:
error = data_copy['increase_stock'].astype('string') == data_copy['y_hat'].astype('string')
error.mean()

0.82

# Choosing always 'low_bike_demand' gives a baseline of 82% accuracy